diff --git a/lib/microx/microx_conqueror.ml b/lib/microx/microx_conqueror.ml
index 05bc5df..7e25961 100644
--- a/lib/microx/microx_conqueror.ml
+++ b/lib/microx/microx_conqueror.ml
@@ -139,6 +139,7 @@ class ['a] main ?(deterministic=false) p =
       | None -> self#break
       | Some ctxt ->
         SM.put ctxt >>= fun () ->
+        let jmp = Jmp.with_cond jmp (Bil.int Word.b1) in
         super#eval_jmp jmp >>= fun () ->
         SM.get () >>= fun ctxt ->
         match ctxt#next with
